src/jobs/anonymizeBeneficiaries.ts

```diff
@@ -3,63 +3,72 @@ import { Types } from 'mongoose';
+  });
 
-    // Get Family record
-    const familyRecord = await Record.findOne({
-      _id: aidRecord?.data?.owner_resource,
+  // For each family record, check if exists
+  // an aid record in the last 18 months
+  for (const family of allFamilies) {
+    const aidGivenToFamily = await Record.exists({
+      resource: AID_RESOURCE_ID,
+      createdAt: {
+        $gt: new Date(Date.now() - 18 * 30 * 24 * 60 * 60 * 1000),
+      }, // 18 months ago
+      'data.owner_resource': family._id.toString(),
     });
 
-    // Find all members of the family
-    const members = await Record.find({
-      _id: { $in: familyRecord?.data?.members },
-    });
+    // If no aid was given to the family in the last 18 months
+    if (!aidGivenToFamily) {
+      // Find all members of the family
+      const members = await Record.find({
+        _id: { $in: family?.data?.members },
+      });
 
-    // Anonymize all members of that family
-    for (let j = 0; j < members.length; j++) {
-      const member = members[j];
+      // Anonymize all the members
+      members.forEach((member) => {
+        if (!member.data) {
+          return;
+        }
+        // Anonymize the member
+        member._createdBy = new User({
+          name: 'ANONYMOUS',
+          username: `${member._id.toString()}@oort-anonymous.com`,
+        });
 
-      // Anonymize the member
-      member._createdBy = new User({
-        name: 'ANONYMOUS',
-        username: `${Math.random()
-          .toString(36)
-          .substring(2, 15)}@anonymus-oort.com`,
-      });
-      member.data.location = 'ANONYMOUS';
-      member.data.surname = 'ANONYMOUS';
-      member.data.firstname = 'ANONYMOUS';
-      member.data.phone = 'ANONYMOUS';
-      member.data.nom_employes = 'ANONYMOUS';
-      member.data.gender = 'ANONYMOUS';
-      member.data.birthdate = 'ANONYMOUS';
-      member.data.prenom_employes = 'ANONYMOUS';
-      member.data.nom_prenom_employes = 'ANONYMOUS';
-      member.data.tel_staff = 'ANONYMOUS';
-      member.data.email_staff = 'ANONYMOUS';
-      member.data.birthdate_employes = 'ANONYMOUS';
-      member._lastUpdatedBy = new User({
-        name: 'ANONYMOUS',
-        username: `${Math.random()
-          .toString(36)
-          .substring(2, 15)}@anonymus-oort.com`,
+        member.data = {
+          ...member.data,
+          location: 'ANONYMOUS',
+          surname: 'ANONYMOUS',
+          firstname: 'ANONYMOUS',
+          phone: 'ANONYMOUS',
+          nom_employes: 'ANONYMOUS',
+          gender: 'ANONYMOUS',
+          birthdate: 'ANONYMOUS',
+          prenom_employes: 'ANONYMOUS',
+          nom_prenom_employes: 'ANONYMOUS',
+          tel_staff: 'ANONYMOUS',
+          email_staff: 'ANONYMOUS',
+          birthdate_employes: 'ANONYMOUS',
+          file_gdpr_staff: [],
+        };
+
+        member._lastUpdatedBy = new User({
+          name: 'ANONYMOUS',
+          username: `${member._id.toString()}@oort-anonymous.com`,
+        });
```

Comment:
Same here, can be removed. And actually, another change you should make is doing this but in the staff anonymization, so when anonymizing a staff, you have to check all records created or last updated by them and updated the _lastUpdatedby and the _createdBy. However, do not create a new user, as that creates a new id, and we do not want that.